csharp/src/Ice/Acm.cs

```diff
@@ -10,6 +10,99 @@
+        /// the connection has pending invocations.</summary>
+        OnInvocation,
+        /// <summary>Combines the behaviors of CloseOnIdle and CloseOnInvocation.</summary>
+        OnInvocationAndIdle,
+        /// <summary>Forcefully closes a connection that has been idle for the configured timeout period, regardless of
+        /// whether the connection has pending invocations or dispatch.</summary>
+        OnIdleForceful
+    }
+
+    /// <summary>Specifies the heartbeat semantics for ACM (Active Connection Management).</summary>
+    public enum AcmHeartbeat
+    {
+        /// <summary>Disables heartbeats.</summary>
+        Off,
+        /// <summary>Send a heartbeat at regular intervals if the connection is idle and only if there are pending
+        /// dispatch.</summary>
+        OnDispatch,
+        /// <summary>Send a heartbeat at regular intervals when the connection is idle.</summary>
+        OnIdle,
+        /// <summary>Send a heartbeat at regular intervals until the connection is closed.</summary>
+        Always
+    }
+
+    /// <summary>This struct represents the Acm (Active Connection Management) configuration.</summary>
+    public readonly struct Acm : IEquatable<Acm>
+    {
+        /// <summary>Gets the Acm configuration for disabling activation connection management.</summary>
+        public static Acm Disabled =>
+            new Acm(System.Threading.Timeout.InfiniteTimeSpan, AcmClose.Off, AcmHeartbeat.Off);
+
+        /// <summary>Gets the <see cref="AcmClose"/> setting for the Acm configuration.</summary>
+        public AcmClose Close { get; }
+        /// <summary>Gets <see cref="AcmHeartbeat"/> setting for the Acm configuration.</summary>
+        public AcmHeartbeat Heartbeat { get; }
+        /// <summary>Returns true if ACM is disabled, false otherwise.</summary>
+        public bool IsDisabled => (Close == AcmClose.Off && Heartbeat == AcmHeartbeat.Off) ||
+            Timeout == System.Threading.Timeout.InfiniteTimeSpan || Timeout == TimeSpan.Zero;
+        /// <summary>Gets the timeout setting for the Acm configuration.</summary>
+        public TimeSpan Timeout { get; }
+
+        internal static Acm ServerDefault =>
+            new Acm(TimeSpan.FromSeconds(60), AcmClose.OnInvocation, AcmHeartbeat.OnDispatch);
+        internal static Acm ClientDefault =>
+            new Acm(TimeSpan.FromSeconds(60), AcmClose.OnInvocationAndIdle, AcmHeartbeat.OnDispatch);
+
+        /// <summary>Creates an Acm configuration structure.</summary>
+        /// <param name="timeout">The timeout setting. The timeout must be a positive, non-zero value.</param>
+        /// <param name="close">The close setting.</param>
+        /// <param name="heartbeat">The heartbeat setting.</param>
+        public Acm(TimeSpan timeout, AcmClose close, AcmHeartbeat heartbeat)
+        {
+            if (timeout != System.Threading.Timeout.InfiniteTimeSpan && timeout < TimeSpan.Zero)
```

Comment:
This check does not look correct.

Reply:
Isn't this just to avoid negative timeouts other than -1 (InfiniteTimeSpan)

Reply:
Right, InfiniteTimeSpan is indeed negative.